src/createWidgetBase.ts

```diff
@@ -197,16 +216,37 @@ const createWidget: WidgetFactory = createStateful
 				});
 			},
 
-			get id(this: Widget<WidgetState>): string {
+			get id(this: Widget<WidgetState, WidgetProperties>): string {
 				const { id } = widgetInternalStateMap.get(this);
 
-				return id || (this.state && this.state.id) || generateID(this);
+				return id;
+			},
+
+			applyChangedProperties: function(this: Widget<WidgetState, WidgetProperties>, previousProperties: Partial<WidgetProperties>, currentProperties: Partial<WidgetProperties>): void {
```

Comment:
So I think with the `id` carry over issue, for now we can just apply the `id` in this block like so:
```typescript
applyChangedProperties: function(this: Widget<WidgetState, WidgetProperties>, previousProperties: Partial<WidgetProperties>, currentProperties: Partial<WidgetProperties>): void {
    if (Object.keys(currentProperties).length) {
        currentProperties.id = this.id;
        this.setState(currentProperties);
    }
}
```